### PR TITLE
Zest subs and saved cards

### PR DESCRIPTION
**Pull request by Mopok**
**Changes**

- Added saved cards to profile page
  - Add cards
  - Update cards
  - Delete cards
screen shot 2017-04-10 at 4 03 21 pm

screen shot 2017-04-09 at 10 57 04 am

- Subscriptions page
  - Lists customers subscriptions
screen shot 2017-04-09 at 12 27 23 pm

- Subscription page
  - List of products in subscription
  - Update payment card for subscription
  - Add payment card if none exist
  - Delete card
  - Orders associated with subscription
  - Billing plan updater
screen shot 2017-04-09 at 12 29 54 pm

- Cart
  - Billing plan dropdown, only shows if a billing plan is set
- Checkout pay
  - Now has default template for spreedly integrated payment methods.
screen shot 2017-04-10 at 2 03 21 pm

- Header
  - Only shows my subscriptions if the customer has subscriptions
  - Has a profile dropdown, contains my orders and my subscriptions(if there is any)

**Notes**
Was branched from Andy's demo branch and cleaned up/modifed from there.

### CODE REVIEW

File: partials/shop-paymentform.htm [hunk 1]
```diff
@@ -1,6 +1,33 @@
 {% if paymentMethod is defined %}
-    {% set name = paymentMethod.getFrontendPartialName() %}
-    {{ partial(name, {paymentMethod: paymentMethod, payment: payment}) }}
+    {% if paymentMethod.isStandard() %}
+        {{ paymentForm({
+                    options: {
+                        number: {
+                            placeholder: 'Card number',
+                            style:"font-family: Courier New, Courier, monospace; letter-spacing: 0px; font-weight: 200; font-size: 16px; color: rgb(58, 61, 63); line-height: 32px; width: 345px;"
```
Comment (robotpony) [inline partials/shop-paymentform.htm]: You need a space after the colon:

`style:"`

Should be:

```
style: "
      ^
```

File: resources/css/style.css [hunk 2]
```diff
@@ -656,6 +658,72 @@ footer p {font-size: 12px;}
...
+    line-height: 32px;
+  }
+  .ls-card-form .card-form-input::-webkit-input-placeholder {
+    font-family: Courier New, Courier, monospace;
+    color: rgb(58, 61, 63) !important;
+    font-weight: 500;
+    font-size: 16px;
+    letter-spacing: 0px;
+  }
+  #submit-button, .ls-card-form .delete-button {
```
Comment (robotpony) [inline resources/css/style.css]: CSS lint guidelines require that each selector gets its own line:

`#submit-button, .ls-card-form .delete-button {`

Should be:

```
#submit-button,
.ls-card-form .delete-button {
```

File: resources/css/style.css [hunk 3]
```diff
@@ -977,3 +1045,22 @@ strong {
     font-size: 0.5em;
     margin: 0.25em 0 1.5em 0;
     color: #555; }
+
+/* Profile */
+  #change-password-container { border-right: 1px solid #e2e2e2; }
```
Comment (robotpony) [inline resources/css/style.css]: CSS lint doesn't allow single-line selectors like this.

File: partials/shop-paymentform.htm [hunk 4]
```diff
@@ -1,6 +1,33 @@
...
+                        full_name: {
+                            placeholder: 'Cardholder name',
+                        },
+                        expiry: {
+                            placeholder: 'MM/YYYY',
+                        },
+                        submit: {
+                            value: 'Pay via Credit Card'
+                        },
+                        save_card: true
```
Comment (aperdomo) [inline partials/shop-paymentform.htm]: This should be changed to:

```
save_card: {
    enabled: true
}
```

My bad for not nudging docs there.

**Comment by aperdomo** [comment]
@Mopok 

Two notes:
- Paying with stored cards still needs to be added.
- I had some issues downloading the zip from this branch and uploading it to my store yesterday. Some pages (product, profile) were hanging. Pulling your working theme from stage and uploading that resolved it. Can you try uploading the zip from this branch to sanity check that the branch theme is clean?

Well done on everything else though. It's looking great.

**Comment by dhalarewich** [comment]
The checkbox on the add card modal (shown in first screenshot): I'd change the textbox to read "Make my default card"

Looking great!

**Comment by Mopok** [comment]
This is ready. One more thing needs to be done in the backend see #4100. Then zest should be good to go. Checkout pay looks as follows:
screen shot 2017-04-13 at 4 32 18 pm
screen shot 2017-04-13 at 4 32 27 pm
screen shot 2017-04-13 at 4 32 54 pm

I also updated the default card text Danny Suggested.

**Pull request merged**